packages/block-library/src/query-loop/edit.js

```diff
@@ -78,6 +80,14 @@ export default function QueryLoopEdit( {
 			if ( sticky ) {
 				query.sticky = sticky === 'only';
 			}
+
+			if ( useGlobalQuery ) {
+				const templateQuery = getTemplateQuery();
```

Comment:
`getTemplateQuery` should either be inlined and use `select` calls from the scope of this function or extracted to another hook that is built on top of `useSelect`.